File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    tangerine.sort()
    typetan = len(set(tangerine))
    countdict = {}
    for i in range(len(tangerine)) :
        if tangerine[i] not in countdict.keys() : 
            countdict[tangerine[i]] = 1
        else:
            countdict[tangerine[i]] +=1 
            
    tangerine.sort(key = lambda x : countdict[x])
  
    length  = len(tangerine)


            
    return len(set(tangerine[length-k:]))

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> mymap = new HashMap<>();
        for (int i = 0; i < tangerine.length; i++) {
            mymap.put(tangerine[i], mymap.getOrDefault(tangerine[i],0)+1);
        }
        
        // 개수 기준 내림차순 정렬
        List<Integer> values = new ArrayList<>(mymap.values());
        Collections.sort(values, Collections.reverseOrder());
        
        int sum = 0;
        int count = 0;
        for (int v : values) {
            sum += v;
            count++;
            if (sum >= k) {
                return count;
            }
        }
        
        return count;
    }
}